retry/lib/retry.dart

```diff
@@ -104,7 +104,7 @@ class RetryOptions {
       return Duration.zero;
     }
     final rf = (randomizationFactor * (_rand.nextDouble() * 2 - 1) + 1);
-    final delay = (delayFactor * math.pow(2, attempt) * rf);
+    final delay = (delayFactor * math.pow(2.0, attempt) * rf);
```

Comment:
Hmm, maybe it would be better to limit the exponent... 
In practice `maxDelay` should always make it less than `delayFactor * 2^31` (I hope :rofl:).

```suggestion
    final exp = math.max(attempt, 31); // prevent overflows.
    final delay = delayFactor * math.pow(2, exp) * rf;
```

Reply:
This might just be a nit, but ints feels more predictable than floats here :D

Reply:
Your code make sense though `math.max` should be `math.min`.
I fix it.

Reply:
Haha, well, at-least I was close, hehe :D